x/projects/keeper/project_test.go

```diff
@@ -45,69 +88,111 @@ func prepareProjectsData(ctx context.Context, keepers *testkeeper.Keepers) (proj
 	}
 
 	templates := []struct {
+		code    string
 		name    string
 		enabled bool
 		keys    []types.ProjectKey
 		policy  *types.Policy
 	}{
 		// project with admin key, enabled, has policy
-		{"mock_project_1", true, keys_1_admin, policy1},
+		{"pd1", "mock_project_1", true, keys_1_admin, policy1},
 		// project with "both" key, disabled, with policy
-		{"mock_project_2", false, keys_1_admin_dev, policy1},
+		{"pd2", "mock_project_2", false, keys_1_admin_dev, policy1},
 		// project with 2 keys (one admin, one developer) disabled, no policy
-		{"mock_project_3", false, keys_2_admin_and_dev, nil},
+		{"pd3", "mock_project_3", false, keys_2_admin_and_dev, nil},
 	}
 
 	for _, tt := range templates {
-		projectData := types.ProjectData{
+		ts.projects[tt.code] = types.ProjectData{
 			Name:        tt.name,
 			Description: "",
 			Enabled:     tt.enabled,
 			ProjectKeys: tt.keys,
 			Policy:      tt.policy,
 		}
-		projects = append(projects, projectData)
 	}
+}
+
+func (ts *testStruct) BlockHeight() uint64 {
+	return uint64(ts.ctx.BlockHeight())
+}
+
+func (ts *testStruct) AdvanceBlock(count int) {
+	for i := 0; i < count; i += 1 {
+		ts._ctx = testkeeper.AdvanceBlock(ts._ctx, ts.keepers)
+	}
+	ts.ctx = sdk.UnwrapSDKContext(ts._ctx)
+}
+
+func (ts *testStruct) AdvanceEpoch(count int) {
+	for i := 0; i < count; i += 1 {
+		ts._ctx = testkeeper.AdvanceEpoch(ts._ctx, ts.keepers)
+	}
+	ts.ctx = sdk.UnwrapSDKContext(ts._ctx)
+}
+
+func (ts *testStruct) isKeyInProject(index, key string, kind types.ProjectKey_Type) bool {
```

Comment:
There will be a basic TestStruct to be shared by all modules - it will have the _ctx, ctx, keepers, servers, and some helpers like AdvanceBlock etc.  The idea is that each module will embed that in a (module-) private e.g. projectTestStruct which will then inherit all the common helpers/goodies, and add the module-specific ones.
In a follow-up PR.